RecordPoint.Connectors.SDK.Client/SubmitPipeline/FilterPipelineElement.cs

```diff
@@ -63,33 +63,25 @@ private MatchResult MatchesFilter(SubmitContext submitContext, SearchTreeNodeMod
 
             // Otherwise, evaluate using the boolean operator and all the children of the SearchTerm
             var results = filter.Children.Select(x => MatchesFilter(submitContext, x, nullResult));
-            switch (filter.BoolOperator)
+            if (filter.BoolOperator.Equals(FilterConstants.FilterBooleanOperators.And.ToString(), StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Nitpick: This is fairly hard to read. Consider making a method for this comparison.